nuxeo-distribution/nuxeo-server-hotreload-tests/src/test/java/org/nuxeo/ftest/server/hotreload/ITDevHotReloadTest.java

```diff
@@ -249,4 +263,60 @@ public void testHotReloadFileImporters() throws IOException {
         assertEquals("Foo", docType);
     }
 
+    @Test
+    /** @since 11.1 **/
+    public void shouldComputeMimeTypeWhenHotReloadDocumentModelSchema() throws IOException {
+        hotReloadRule.deployJarDevBundle(ITDevHotReloadTest.class, "testHotReloadSchema");
+
+        // Creates the document.
+        String docId = RestHelper.createDocument("/", "CustomDocType", "myDocument");
+        Document document = RestHelper.repository().fetchDocumentById(docId);
+
+        // Creates the contents.
+        BatchUploadManager batchUploadManager = RestHelper.batchUploadManager();
+
+        FileBlob image = new FileBlob(Files.createTempFile("", "").toFile(), "anyImage.png");
+        BatchUpload imageBatch = batchUploadManager.createBatch();
+        imageBatch = imageBatch.upload("0", image);
+
+        FileBlob audio = new FileBlob(Files.createTempFile("", "").toFile(), "anySong.mp3");
+        BatchUpload audioBatch = batchUploadManager.createBatch();
+        audioBatch = audioBatch.upload("0", audio);
+
+        // Attach the contents to the document.
+        document.setPropertyValue(CUSTOM_IMAGE_PROPERTY_PATH, imageBatch.getBatchBlob());
+        document.setPropertyValue("custom:question",
+                Map.of("audio", audioBatch.getBatchBlob(), "image", imageBatch.getBatchBlob()));
+        document.updateDocument();
+
+        // Checks that all mime types are correctly computed.
+        document = RestHelper.repository().header("properties", "*").fetchDocumentById(docId);
```

Comment:
Here you can introduce a method like `fetchDirectoryEntryProperties` in helper.
Note: there's an API to tell the java client which properties it needs to fetch, see https://doc.nuxeo.com/client-java/3.2/#client-creation